searx/network/network.py

```diff
@@ -167,9 +167,6 @@ async def check_tor_proxy(client: httpx.AsyncClient, proxies) -> bool:
         for transport in client._mounts.values():  # pylint: disable=protected-access
             if isinstance(transport, AsyncHTTPTransportNoHttp):
                 continue
-            if not getattr(transport, '_rdns', False):
```

Comment:
Do you have an example to reproduce the bug on this line?
With:
```yaml
outgoing:
  using_tor_proxy: true
```

I have these errors
```ERROR   searx.network.image_proxy     : Error
Traceback (most recent call last):
  File "/home/alexandre/code/searxng/searx/network/network.py", line 289, in check
    await network.get_client()
  File "/home/alexandre/code/searxng/searx/network/network.py", line 205, in get_client
    raise httpx.ProxyError('Network configuration problem: not using Tor')
httpx.ProxyError: Network configuration problem: not using Tor
Traceback (most recent call last):
  File "/home/alexandre/code/anaconda3/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/alexandre/code/anaconda3/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/alexandre/code/searxng/searx/webapp.py", line 1361, in <module>
    search_initialize(enable_checker=True, check_network=True, enable_metrics=settings['general']['enable_metrics'])
  File "/home/alexandre/code/searxng/searx/search/__init__.py", line 32, in initialize
    check_network_configuration()
  File "/home/alexandre/code/searxng/searx/network/network.py", line 298, in check_network_configuration
    raise RuntimeError("Invalid network configuration")
RuntimeError: Invalid network configuration
```

It is expected.

Reply:
If you set `using_tor_proxy: true` in `outgoing` you can only use it if everything is routed over Tor.
If got the same error as you do.
What do you mean by expected?

Reply:
`using_tor_proxy: true` means that requests are sent using tor.
To avoid leak, SearXNG (and searx) checks if the HTTP requests are sent using tor. If this is not the case, the SearxNG does not start.

Even if I remove the `if not getattr(transport, '_rdns', False):` checks, I still get the same error.

---

The `_rdns` check is related to this configuration (inherit from [requests](https://docs.python-requests.org/en/latest/)):
* `socks5://127.0.0.1:9050` tells to use the local DNS --> `_rdns = False`. This is not what we want for tor.
* `socks5h://127.0.0.1:9050` tells to use the DNS from the proxy --> `_rdns = True`. This is what we want for tor.

(rdns means "remote DNS")

Out of topic: I wonder if this legacy is required, I mean, is there a valid configuration with a proxy and local DNS resolution.

Reply:
If you set a Tor engine to use a socks5 proxy without name resolution at the proxy in SearXNG it won't be able to contact the onion website. Tor doesn't use normal DNS for name resolution, you need a Hidden Service Directory for that.
I think I don't understand what your arguments against checking if a Tor proxy is used for each engine vs only globally.